samplesheets/views_api.py

```diff
@@ -310,6 +323,240 @@ def post(self, request, *args, **kwargs):
+                        build_secret(16),
+                        serializer.validated_data.get('date_expires'),
+                    )
+            except Exception as ex:
+                raise ValidationError(
+                    '{} {}'.format('Creating ' + IRODS_TICKET_EX_MSG + ':', ex)
+                )
+
+            # Create database object
+            serializer.validated_data['ticket'] = ticket.ticket
+            irods_access_ticket = serializer.save()
+
+            # Create timeline event and app alerts
+            self.create_timeline_event(irods_access_ticket, 'create')
+            self.create_app_alerts(
+                irods_access_ticket, 'create', self.request.user
+            )
+        else:
+            raise ValidationError('Invalid data: {}'.format(serializer.errors))
+        return Response(serializer.data, status=status.HTTP_201_CREATED)
+
+
+class IrodsAccessTicketUpdateAPIView(
+    IrodsAccessTicketModifyMixin, SODARAPIBaseProjectMixin, UpdateAPIView
+):
+    """
+    Update an iRODS access ticket for a project.
+
+    **URL:** ``/samplesheets/api/irods/ticket/update/{IrodsAccessTicket.sodar_uuid}``
+
+    **Methods:** ``PUT``, ``PATCH``
+
+    **Parameters:**
+
+    - ``label``: Label
+    - ``date_expires``: Expiration date (YYYY-MM-DD)
+
+    **Returns:**
+
+    - ``ticket``: iRODS access ticket
+    """
+
+    permission_required = 'samplesheets.edit_sheet'
+
+    def get_object(self):
+        return IrodsAccessTicket.objects.get(
+            sodar_uuid=self.kwargs.get('irodsaccessticket')
+        )
+
+    def put(self, request, *args, **kwargs):
+        """PUT request for updating an iRODS access ticket"""
+        ticket = self.get_object()
+        if request.data.keys() - {'label', 'date_expires'}:
+            raise ValidationError(
+                '{} {}'.format(
+                    'Updating ' + IRODS_TICKET_EX_MSG + ':',
+                    IRODS_TICKET_NO_UPDATE_FIELDS_MSG
+                    + ' '
+                    + ', '.join(
+                        request.data.keys() - {'label', 'date_expires'}
+                    ),
+                )
+            )
+        serializer = IrodsAccessTicketSerializer(
+            ticket, data=request.data, partial=True
+        )
+        serializer.context['project'] = self.get_project()
+
+        if serializer.is_valid():
+            serializer.save()
+        else:
+            raise ValidationError(
+                'Updating {}: {}: {}'.format(
+                    IRODS_TICKET_EX_MSG,
+                    IRODS_TICKET_INVALID_DATA_MSG,
+                    serializer.errors,
+                )
+            )
+        self.create_timeline_event(ticket, 'update')
+        self.create_app_alerts(ticket, 'update', self.request.user)
+        return Response(serializer.data, status=status.HTTP_200_OK)
+
+
+class IrodsAccessTicketDestroyAPIView(
+    IrodsAccessTicketModifyMixin, SODARAPIBaseProjectMixin, DestroyAPIView
```

Comment:
This should have `SODARAPIGenericProjectMixin` instead of `SODARAPIBaseProjectMixin`.